### ISSUE TITLE
Crash when loading a form with invalid `itemsets.csv`

### ISSUE THREAD

**Issue by seadowg** [getodk/collect#5044]
#### Software and hardware versions 

Collect v2022.1

#### Problem description

Collect crashes when loading an `itemsets.csv` where a row has an extra column:

```csv
list_name, name, label
plays, twelfth, Twelfth Night
plays, tempest, The Tempest, EXTRA
plays, macbeth, The Scottish Play
```

Here the app would crash when loading the form but would then load fine the second time. Worse still, there is no error shown, and the form will be missing choices including and after the row with the extra column. For instance, in the above example, the only choice available to the form would be `Twelfth Night`.

It is easy to create this situation with an invalid form that includes a delimiter in an unquoted value:

```
```csv
list_name, name, label
addresses, home, Drury Lane, England
```

The `home` row would get parsed as 4 columns here (rather than 3).

#### Steps to reproduce the problem

1. Load the [attached form](https://github.com/getodk/collect/files/8152672/selectOneExternal.zip) into Collect
2. Open the form

The app will crash but will open the second time you try to open the form. You'll see that choices are missing (from comparing with `itemsets.csv`).

#### Expected behavior

We should most likely fail to load the form rather than crashing and make sure that the itemset DB is cleaned up to prevent the app from loading the second time with incorrect choices.

#### Other information 

The crash occurs here: https://github.com/getodk/collect/blob/c60d91ba67645465edc973021b5144631e0231f5/collect_app/src/main/java/org/odk/collect/android/fastexternalitemset/ItemsetDbAdapter.java#L137

There is a Firebase report for the crash here: https://console.firebase.google.com/u/0/project/api-project-322300403941/crashlytics/app/android:org.odk.collect.android/issues/bfbb97ba910618878d6ceb568d8ec15c?time=last-seven-days&sessionEventKey=621C99D8028D000172A5907E33AE15B5_1648140557352712133/.

**Comment by lognaturel** [comment]
I think it's good to have this filed for reference but I don't think we want to further support this feature. When we did a round of analytics on it in 2018 it already wasn't used much. Hopefully it's used even less now. My preference would be to tag as `won't fix` and close unless it's 30mins or less of effort.

**Comment by seadowg** [comment]
> I think it's good to have this filed for reference but I don't think we want to further support this feature.

Agreed. My worry here is that it's possible for the issue to fly under the radar (due to the crash without an error and then the form loading the second time) and result in enumerators having less choices displayed than the form designer expects. I think if we're going to mark this as `won't fix` (which I'm in favour of) we should look at marking the` select_one_external` as deprecated, so it's clear there are known problems etc.
